docs/modules/ROOT/pages/tmail-backend/jmap-extensions/pushWithFirebase.adoc

```diff
@@ -0,0 +1,210 @@
+as push subscriptions are not tied to specific accounts. It also does **not** take an *ifInState* argument or return
+*oldState* or *newState* arguments.
+
+The client may update the *expires* property to extend (or, less commonly, shorten) the lifetime of a push subscription.
+The server MAY modify the proposed new expiry time to enforce server-defined limits.
+
+Clients SHOULD NOT update or destroy a push subscription that they did not create (i.e., has a *deviceClientId* that
+they do not recognise).
+
+=== Example
+
+At `2018-07-06T02:14:29Z`, a client with deviceClientId `a889-ffea-910` fetches the set of FirebaseRegistration
+currently on the server, making an API request with:
+
+....
+    [[ "FirebaseRegistration/get", {
+      "ids": null
+    }, "0" ]]
+....
+
+Which returns:
+
+....
+    [[ "FirebaseRegistration/get", {
+      "list": [{
+          "id": "e50b2c1d-9553-41a3-b0a7-a7d26b599ee1",
+          "deviceClientId": "b37ff8001ca0",
+          "expires": "2018-07-31T00:13:21Z",
+          "types": [ "Todo" ]
+      }, {
+          "id": "f2d0aab5-e976-4e8b-ad4b-b380a5b987e4",
+          "deviceClientId": "X8980fc",
+          "expires": "2018-07-12T05:55:00Z",
+          "types": [ "Mailbox", "Email", "EmailDelivery" ]
+      }],
+      "notFound": []
+    }, "0" ]]
+....
+
+Since neither of the returned FirebaseRegistration objects have the client's deviceClientId, it knows it does not have a
+current FirebaseRegistration active on the server. So it creates one, sending this request:
+
+....
+    [[ "FirebaseRegistration/set", {
+      "create": {
+        "4f29": {
+          "deviceClientId": "a889-ffea-910",
+          "token": "firebase-token",
+          "types": null
+        }
+      }
+    }, "0" ]]
+....
+
+The server creates the FirebaseRegistration but limits the expiry time to 7 days in the future, returning this response:
+
+....
+    [[ "FirebaseRegistration/set", {
+      "created": {
+        "4f29": {
+          "id": "P43dcfa4-1dd4-41ef-9156-2c89b3b19c60",
+          "expires": "2018-07-13T02:14:29Z"
+        }
+      }
+    }, "0" ]]
+....
+
+Two days later, the client updates the subscription to extend its lifetime, sending this request:
+
+....
+    [[ "FirebaseRegistration/set", {
+      "update": {
+        "P43dcfa4-1dd4-41ef-9156-2c89b3b19c60": {
+          "expires": "2018-08-13T00:00:00Z"
+        }
+      }
+    }, "0" ]]
+...
+
+The server extends the expiry time, but only again to its maximum limit of 7 days in the future, returning this response:
+
+....
+    [[ "PushSubscription/set", {
+      "updated": {
+        "P43dcfa4-1dd4-41ef-9156-2c89b3b19c60": {
+          "expires": "2018-07-15T02:22:50Z"
+        }
+      }
+    }, "0" ]]
+....
+
+The client then destroy the FirebaseRegistration:
+
+....
+    [[ "FirebaseRegistration/set", {
+      "destroy": ["P43dcfa4-1dd4-41ef-9156-2c89b3b19c60"]
+      }
+    }, "0" ]]
+...
+
+The server confirms the FirebaseRegistration to be destroyed:
```

Comment:
When we have new `Token` in Firebase, we MUST to destroy the old one before create the new one?

Reply:
Yes